src/main/java/org/jenkinsci/plugins/workflow/libs/LibraryAdder.java

```diff
@@ -311,29 +297,25 @@ static List<URL> retrieve(@NonNull LibraryRecord record, @NonNull LibraryRetriev
             if (action != null) {
                 FilePath libs = new FilePath(run.getRootDir()).child("libs");
                 for (LibraryRecord library : action.getLibraries()) {
-                    FilePath libResources = libs.child(library.getDirectoryName() + "/resources/");
-                    FilePath f = libResources.child(name);
-                    if (!new File(f.getRemote()).getCanonicalFile().toPath().startsWith(new File(libResources.getRemote()).getCanonicalPath())) {
-                        throw new AbortException(name + " references a file that is not contained within the library: " + library.name);
```

Comment:
Still needs to be done, except in `lib2Jar`. Unfortunately https://github.com/jenkinsci/jenkins/commit/71d2ecf1a4e5303e80815eaa3935c4f2fa3d9104 + https://github.com/jenkinsci/jenkins/pull/5187 left the critical method restricted to core usage only.